crates/ide_assists/src/handlers/extract_function.rs

```diff
@@ -1177,7 +1177,18 @@ impl FlowHandler {
                 let action = action.make_result_handler(None);
                 let stmt = make::expr_stmt(action);
                 let block = make::block_expr(iter::once(stmt.into()), None);
-                let condition = make::condition(call_expr, None);
+                let controlflow_break_path = make::path_from_text("ControlFlow::Break");
+                let tuple_pat = make::tuple_pat(iter::empty());
+                let condition = make::condition(
+                    call_expr,
+                    Some(
+                        make::tuple_struct_pat(
+                            controlflow_break_path,
+                            iter::once(tuple_pat.into()),
```

Comment:
Generating a `ControlFlow::Break(_)` pattern seems nicer to me

That probably needs a `make::placeholder_pat` version of `make::placeholder_ty`

Reply:
Sounds good. I'll change the pattern to `ControlFlow::Break(_)`.